pkg/controllers/clusterresourceplacement/placement_status.go

```diff
@@ -0,0 +1,312 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package clusterresourceplacement
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	workv1alpha1 "sigs.k8s.io/work-api/pkg/apis/v1alpha1"
+
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	workapi "go.goms.io/fleet/pkg/controllers/work"
+	"go.goms.io/fleet/pkg/utils"
+	"go.goms.io/fleet/pkg/utils/annotations"
+	"go.goms.io/fleet/pkg/utils/controller"
+	"go.goms.io/fleet/pkg/utils/labels"
+)
+
+var (
+	// We only include 100 failed resource placements even if there are more than 100.
+	maxFailedResourcePlacementLimit = 100
+)
+
+const (
+	// ClusterResourcePlacementStatus condition reasons
+	schedulingUnknownReason = "Scheduling"
+
+	synchronizedPendingReason   = "SynchronizedPending"
+	synchronizedSucceededReason = "SynchronizedSucceeded"
+
+	// ApplyFailedReason is the reason string of placement condition when the selected resources fail to apply.
+	ApplyFailedReason = "ApplyFailed"
+	// ApplyPendingReason is the reason string of placement condition when the selected resources are pending to apply.
+	ApplyPendingReason = "ApplyPending"
+	// ApplySucceededReason is the reason string of placement condition when the selected resources are applied successfully.
+	ApplySucceededReason = "ApplySucceeded"
```

Comment:
I wonder why these 3 reasons are exported while others don't.

Reply:
They're used by the integration tests. We could expose others if needed.